server/api/browse.ts

```diff
@@ -0,0 +1,46 @@
+import express from "express";
+import fs from "fs";
+import path from "path";
+
+import { BrowserItem } from "../../common/types";
+import { getHumanSize } from "../../common/utils";
+
+const basePath = path.join(__dirname, "../../");
+const browseApi = express.Router();
+
+const createItem = (name: string, stats: fs.Stats, items?: BrowserItem[]): BrowserItem => {
+  const isDirectory = stats.isDirectory();
+  return {
+    name,
+    sizeBytes: isDirectory ? undefined : stats.size,
+    sizeHuman: isDirectory ? undefined : getHumanSize(stats.size),
+    type: isDirectory ? "dir" : "file",
+    ...(items && { items }),
```

Comment:
It would work, of course; I did it this way for the sake of tests and the feel of the output. Seems more naturally to simply _not have_ the `items` rather than `items: undefined`.

Reply:
hm...if both share the same structure (file/folder) perhaps we can ensure that `items` always exist and that it's an empty array? Dropping this field completely seems a bit strange as it still has `type: file` parameter (should we drop it too?)

Reply:
It's just by looking at `createItem` function alone, it seems it can create various JSON structures for combination of `type` and `items`, so we can have:

1. 
```
type: 'file',
items: []
```

2.
```
type: 'file',
```
3.
```
type: 'dir',
```

4.
```
type: 'dir',
items: []
```

This is minor, but I would add a little bit of consistency

Reply:
Thought about the exact same thing right after my initial reply 👍  I added some consistency by [splitting the item into `BrowserFile` and `BrowserDirectory`](https://github.com/tadeaspetak/teleport-challenge/blob/b58a83c92e3216a854d99c60243a772b4a983280/common/types.ts#L1).

As for `items` on the `BrowserDirectory`, I think it would be nice if they _could_ be `undefined`. Right now, you only get 1-level deep contents of a directory. In the future, however, you might be able to get more levels or an entire file tree down from a certain point. `items: []` would then indicate an empty directory whereas `undefined` that, at the given point in time, we simply don't know yet.